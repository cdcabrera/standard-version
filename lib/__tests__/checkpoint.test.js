const checkpoint = require('../checkpoint');

describe('checkpoint', () => {
  it('should attempt to display console messaging', () => {
    const mockInfo = jest.fn();
    const spy = jest.spyOn(console, 'info').mockImplementation(mockInfo);

    const args = ['lorem ipsum'];
    args.dryRun = true;
    checkpoint({ silent: false }, undefined, args, undefined);

    args.dryRun = false;
    checkpoint({ silent: false }, undefined, args, undefined);
    checkpoint({ silent: false }, undefined, args, 'mock-figure');
    checkpoint({ silent: false }, 'dolor sit', args, 'mock-figure');

    expect(mockInfo.mock.calls).toMatchSnapshot('console');

    spy.mockClear();
  });
});
