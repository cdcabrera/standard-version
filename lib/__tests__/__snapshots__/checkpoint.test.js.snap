// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkpoint should attempt to display console messaging: console 1`] = `
Array [
  Array [
    "<yellow>✔</yellow> undefined <bold>lorem ipsum</bold>",
  ],
  Array [
    "<green>✔</green> undefined <bold>lorem ipsum</bold>",
  ],
  Array [
    "mock-figure undefined <bold>lorem ipsum</bold>",
  ],
  Array [
    "mock-figure dolor sit <bold>lorem ipsum</bold>",
  ],
]
`;
